### ISSUE TITLE
Loss is not decreasing during training

### ISSUE THREAD

**Issue by generalsvr** [a-r-r-o-w/finetrainers#157]
### System Info / 系統信息

- 🤗 Diffusers version: 0.33.0.dev0
- Platform: Linux-6.5.0-45-generic-x86_64-with-glibc2.35
- Running on Google Colab?: No
- Python version: 3.11.10
- PyTorch version (GPU?): 2.5.1+cu124 (True)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Huggingface_hub version: 0.27.0
- Transformers version: 4.47.1
- Accelerate version: 1.2.1
- PEFT version: 0.14.0
- Bitsandbytes version: 0.45.0
- Safetensors version: 0.4.5
- xFormers version: not installed
- Accelerator: NVIDIA A100 80GB PCIe, 81920 MiB
NVIDIA A100 80GB PCIe, 81920 MiB

### Information / 问题信息

- [X] The official example scripts / 官方的示例脚本
- [ ] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

When running training using demo dataset Wild-Heart/Disney-VideoGeneration-Dataset, HunyuanVideo seems not to learn. Tried with my own dataset also, same result. I am using original bash script with accelerate_configs/deepspeed.yaml config. Here is training log:

```
12/27/2024 17:51:03 - INFO - finetrainers - Training configuration: {
    "trainable parameters": 163577856,
    "total samples": 69,
    "train epochs": 1,
    "train steps": 25,
    "batches per device": 1,
    "total batches observed per epoch": 35,
    "train batch size": 2,
    "gradient accumulation steps": 1
}

loss_values = [
    0.0285,  # step 1
    0.201,   # step 2
    0.035,   # step 3
    0.0902,  # step 4
    0.055,   # step 5
    0.0406,  # step 6
    0.0368,  # step 7
    0.0474,  # step 8
    0.0727,  # step 9
    0.0752,  # step 10
    0.0323,  # step 11
    0.211,   # step 12
    0.0824,  # step 13
    0.0478,  # step 14
    0.0352,  # step 15
    0.0382,  # step 16
    0.0393   # step 17
]

grad_norm_values = [
    0.00135,  # step 1
    0.00405,  # step 2
    0.00172,  # step 3
    0.00565,  # step 4
    0.00141,  # step 5
    0.00303,  # step 6
    0.00274,  # step 7
    0.00380,  # step 8
    0.00563,  # step 9
    0.00502,  # step 10
    0.00149,  # step 11
    0.00409,  # step 12
    0.00268,  # step 13
    0.00335,  # step 14
    0.00148,  # step 15
    0.00165,  # step 16
    0.00229   # step 17
]

lr_values = [
    2e-7,    # step 1
    4e-7,    # step 2
    6e-7,    # step 3
    8e-7,    # step 4
    1e-6,    # step 5
    1.2e-6,  # step 6
    1.4e-6,  # step 7
    1.6e-6,  # step 8
    1.8e-6,  # step 9
    2e-6,    # step 10
    2.2e-6,  # step 11
    2.4e-6,  # step 12
    2.6e-6,  # step 13
    2.8e-6,  # step 14
    3e-6,    # step 15
    3.2e-6,  # step 16
    3.4e-6   # step 17
]
```

Running inference using original script with local lora:

```
import torch
from diffusers import HunyuanVideoPipeline

import torch
from diffusers import HunyuanVideoPipeline, HunyuanVideoTransformer3DModel
from diffusers.utils import export_to_video

model_id = "hunyuanvideo-community/HunyuanVideo"
transformer = HunyuanVideoTransformer3DModel.from_pretrained(
    model_id, subfolder="transformer", torch_dtype=torch.bfloat16
)
pipe = HunyuanVideoPipeline.from_pretrained(model_id, transformer=transformer, torch_dtype=torch.float16)

pipe.load_lora_weights("/workspace/models/checkpoint-50/pytorch_lora_weights.safetensors", weight_name="default_0",local_files_only=True)
pipe.set_adapters(["default_0"], [0.9])

pipe.vae.enable_tiling()
pipe.to("cuda")

output = pipe(
    prompt="Prompt here",
    height=320,
    width=512,
    num_frames=61,
    num_inference_steps=30,
).frames[0]
export_to_video(output, "output1.mp4", fps=15)
```

### Expected behavior / 期待表现

Expected to get trained model, what am I doing wrong?

**Comment by a-r-r-o-w** [comment]
This is what a typical diffusion loss curve looks like:

![image](https://github.com/user-attachments/assets/961a7ee8-2cd9-481e-9414-3671d61387b3)

You should be able to see a clearer decreasing trend if you plot loss for a particular timestep. Currently, the plot is for training step vs loss. A training step can have any uniformly random sampled timestep value between 0 and 1000, so the values that you see are not really representative of the model learning to better on your data

**Comment by generalsvr** [comment]
> This is what a typical diffusion loss curve looks like:
> 
> ![image](https://private-user-images.githubusercontent.com/72266394/399001561-961a7ee8-2cd9-481e-9414-3671d61387b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzNTk4MTksIm5iZiI6MTczNTM1OTUxOSwicGF0aCI6Ii83MjI2NjM5NC8zOTkwMDE1NjEtOTYxYTdlZTgtMmNkOS00ODFlLTk0MTQtMzY3MWQ2MTM4N2IzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI4VDA0MTgzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE3YTg3YTM3OTc2ZDNlYWVmMTA5MTMyY2Y0NTVmMjQyZmY1NDI5ZmNkYzIyMGIwOTg4MjVkNmMwNTYxOWViOTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OHz4AUFk-WldcM2Z-0wKNk7WJtY1E7eXzz3auRJiyeQ)
> 
> You should be able to see a clearer decreasing trend if you plot loss for a particular timestep. Currently, the plot is for training step vs loss. A training step can have any uniformly random sampled timestep value between 0 and 1000, so the values that you see are not really representative of the model learning to better on your data

So that means I just need to train longer? What is the number of steps in your experience works the best for a small dataset 20-40 videos? I can't see positive results after 50 steps on my 20 vid dataset. Models looks like original

**Comment by a-r-r-o-w** [comment]
Not really. Training longer will still give you a similar looking loss curve. But, "the length of bumps" that you see should be getting lower on very long training runs. This randomness comes from the fact that timesteps are randomly sampled during the training process and the model gradually learns how to work with your data over time for all timesteps, but until it does, it will look fully random like above.

For 20-40 videos of a similar concept, I would recommend 2000-10000 steps depending on the difficulty. Character loras usually require more steps compared to style loras, so you will have to experiment a bit. One good way to do this is enable checkpointing every 500 steps (by specifying `--checkpointing_steps 500`, and just setting training steps to 1 million. When you feel your model has learned well enough, you can just force quit the training run. To do this, you would usually use a constant learning rate schedule and low values for lr (such as `1e-5` or lower).

LMK if that helps :)
